test/public/CreateX.deployCreate.tree

```diff
@@ -0,0 +1,9 @@
+CreateX_DeployCreate_Public_Test
+├── When the initCode creates a valid runtime bytecode
+│   ├── When the created runtime bytecode has a non-zero length
+│   │   ├── It returns a contract address with a non-zero bytecode length and a potential non-zero ether balance.
+│   │   └── It emits the event ContractCreation with the contract address as indexed argument.
+│   └── When the created runtime bytecode has a zero length
+│       └── It should revert.
+└── When the contract creation fails
+    └── It should revert.
```

Comment:
wdyt of flattening this a bit to remove the very top `When`, resulting in:
- When the initCode creates runtime bytecode with a non-zero length
- When the initCode creates runtime bytecode with zero length
- When the initCode fails to deploy runtime bytecode

Reply:
Yeah, I think that makes sense. Implemented via ca8429d09ec5a241d42d932e5be61bfcf8cbadd9.